postgres/src/main/scala/zio/sql/postgresql/PostgresModule.scala

```diff
@@ -2,12 +2,36 @@ package zio.sql.postgresql
 
 import zio.sql.Jdbc
 
+import scala.language.implicitConversions
+
 /**
  */
 trait PostgresModule extends Jdbc { self =>
+  override type ExprExtensionType[F, -A, B] = PostgresExprExtension[F, A, B]
+
+  sealed trait PostgresExprExtension[F, -A, B]
+
+  object PostgresExprExtension {
+    implicit def postgresExpr2Expr[F, A, B: TypeTag](pgExpr: PostgresExprExtension[F, A, B]) =
+      Expr.ExprDialectSpecific(pgExpr)
+
+    implicit def postgresExpr2Selection[F, A, B: TypeTag](
+      pgExpr: PostgresExprExtension[F, A, B]
+    ): Selection[F, A, SelectionSet.Cons[A, B, SelectionSet.Empty]] =
+      Selection.computedOption(pgExpr, Expr.exprName(pgExpr))
 
-  object PostgresFunctionDef {
-    val Sind = FunctionDef[Double, Double](FunctionName("sind"))
+    sealed case class Overlay[F1, F2, F3, F4, Source, String](
+      main: Expr[F1, Source, String],
+      replacing: Expr[F2, Source, String],
+      starting: Expr[F3, Source, Int],
+      number: Expr[F4, Source, Int]
+    ) extends PostgresExprExtension[F1 :||: F2 :||: F3 :||: F4, Source, String] {
+      val functionName = "overlay"
+    }
+  }
+  object PostgresFunctionDef   {
+    val Sind    = FunctionDef[Double, Double](FunctionName("sind"))
+    val Overlay = PostgresExprExtension.Overlay
```

Comment:
Isn't this a problem that there's no indication that Overlay is a FunctionDef?

Reply:
It's not a `FunctionDef`, at least in terms of how `FunctionDef` is defined. `Expr` is our concept of a function A => B with some features F, `FunctionDef` is just a syntax class to allow you easily create `FunctionN` case classes for simple functions, which are themselves just extending `InvariantExpr`

Reply:
a couple of options here:
1. simply rename `PostgresExprExtension` to `PostgresFunction` (since Expr and Function are both functions anyway)
2. we could maybe make `FunctionCall`N traits (not sealed) which could  be self typed as `Expr`s (making it effectively still sealed). We'd then have to come change the existing `FunctionCall`N classes to something like `BasicFunctionCall2 extends InvariantExpr[F1 :||: F2], A, Z] with FunctionCall2[F1, F2 , A, Z] and then we could upcast all the return types to `FunctionCall`N for a nice standardised list of types in autocomplete - the user wouldn't even know which ones are dialect specific... I'm not sure if this is a good thing or not